### ISSUE TITLE
Use sigmajs for realtime monitoring ?

### ISSUE THREAD

**Issue by melvilgit** [jacomyal/sigma.js#659]
Can i update nodes/edges of an existing graph dynamically in sigma.js ? Suppose i need to update color of a nodes in graph. is it possible ?

**Comment by Yomguithereal** [comment]
Yes you can. Just update said node and call `sigInst.refresh();`.

**Comment by melvilgit** [comment]
Hi Yomguithereal. Can i update a single node/edge without updating the entire canvas ?

**Comment by Yomguithereal** [comment]
No. This is not possible. Updating a single node will repaint the canvas. Repainting only some part locally would be as costly as repainting everything in terms of computation for 99% of cases.

**Comment by melvilgit** [comment]
Cool..just for curiousity what is the order of computation for repainting the entire canvas?

**Comment by Yomguithereal** [comment]
Hello @melvilgit. Sorry for the delay. Approx ~O(n), being the number of currently displayed nodes & edges.

Typically, some use cases such as changing only the color of the node could be handled with O(1) but those cases are rare and need to be handled through custom rendering logic to remain viable (else you would have to perform object diff which remains costly).

**Comment by melvilgit** [comment]
Thanks a lot.
I have one more doubt. How can i add arrow attributes in sigma js.

s.graph.edges().forEach(function(edge){

       edge.type = "arrow";


    });

will create arrow heads. But that is not clearly visible.! how can i
increase the size of arrow heads?

On Fri, Oct 16, 2015 at 5:53 PM, Guillaume Plique <notifications@github.com>
wrote:

> Hello @melvilgit <https://github.com/melvilgit>. Sorry for the delay.
> Approx ~O(n), being the number of currently displayed nodes & edges.
>
> Typically, some use cases such as changing only the color of the node
> could be handled with O(1) but those cases are rare and need to be handled
> through custom rendering logic to remain viable (else you would have to
> perform object diff which remains costly).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jacomyal/sigma.js/issues/659#issuecomment-148703189>.
>



-- 
*--------------------*
*skype-melvilsk*
*9611691444*

*Inmobi*
*Bangalore*

**Comment by Yomguithereal** [comment]
@melvilgit, you should try to fiddle with the `minArrowSize` and `maxArrowSize` settings of your instance.

**Comment by melvilgit** [comment]
i have tried these.

s.graph.edges().forEach(function(edge){

       edge.type = "curvedArrow",

       minArrowSize = 25;



    });



even that didn't worked

On Mon, Oct 26, 2015 at 3:26 PM, Guillaume Plique <notifications@github.com>
wrote:

> @melvilgit <https://github.com/melvilgit>, you should try to fiddle with
> the minArrowSize and maxArrowSize settings of your instance.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jacomyal/sigma.js/issues/659#issuecomment-151085281>.
>



-- 
*--------------------*
*skype-melvilsk*
*9611691444*

*Inmobi*
*Bangalore*

**Comment by Yomguithereal** [comment]
In your example, you only seem to initialize a global variable called `minArrowSize`. This won't do anything beyond polluting your global scope. Try the following when instantiating sigma:

```js
var s = new sigma({container: 'whatever', settings: {minArrowSize: 25}});
```

**Comment by melvilgit** [comment]
even that also didn't worked. iam using sigma   Version: 1.0.2. is it have
to do anything ?



On Mon, Oct 26, 2015 at 3:37 PM, Guillaume Plique <notifications@github.com>
wrote:

> In your example, you only seem to initialize a global variable called
> minArrowSize. This won't do anything beyond polluting your global scope.
> Try the following when instantiating sigma:
>
> var s = new sigma({container: 'whatever', settings: {minArrowSize: 25}});
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jacomyal/sigma.js/issues/659#issuecomment-151088521>.
>



-- 
*--------------------*
*skype-melvilsk*
*9611691444*

*Inmobi*
*Bangalore*

**Comment by Yomguithereal** [comment]
Try with this repository version. I don't remember whether v1.0.2 had this option yet.

**Comment by melvilgit** [comment]
it worked .is this new version splitted main sigma.min.js to a no of files
based on functionalit?

On Tue, Oct 27, 2015 at 12:12 AM, Guillaume Plique <notifications@github.com
> wrote:

> Try with this repository version. I don't remember whether v1.0.2 had this
> option yet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jacomyal/sigma.js/issues/659#issuecomment-151243887>.
>



-- 
*--------------------*
*skype-melvilsk*
*9611691444*

*Inmobi*
*Bangalore*

**Comment by Yomguithereal** [comment]
I don't understand what you mean @melvilgit. Could you please reformulate?

**Comment by ygarciac** [comment]
Hi,

I have a similar problem with refresh method. I have done my code to include the nodes and the edges with a for loop. Each iteration, I push my edge on the graph, and then I used the refresh method over sigma instance.

The problem is that when I run my HTML code, I don't see the refresh, but when I execute my code adding a breakpoint , I can see how the graph draw each edge at each iteration.

My number of nodes is 3669 and my number of edges is 3961.

I don't think that the amount of data damage the execution.

If any can help me... Thank you so much!!

My code is this:
`var sigInstance = new sigma({
  container: 'graph-container',
});

setTimeout(function() {
		empezar(0,50);
	}, 5000);
	
setTimeout(function() {
	empezar(50,1000);
	}, 5000);

function empezar(a,b)
{
	for(var i=a; i<b;i+=1)
	{
		edgeAhora = edges[i];
		source = edgeAhora.source;
		target = edgeAhora.target;
		var sourceNode = findNodes(nodes,source);
		var targetNode = findNodes(nodes,target);
		build(sourceNode,targetNode);
	}
}

function build(sourceNode,targetNode)
{	
	var esCiertoSource = contains(existingNodes,sourceNode);
	var esCiertoTarget = contains(existingNodes,targetNode);
		if(esCiertoSource == false){
			sigInstance.graph.addNode(sourceNode);
			existingNodes.push(sourceNode);
		}
		if(esCiertoTarget == false){
			sigInstance.graph.addNode(targetNode);
			existingNodes.push(targetNode);
		}
		sigInstance.graph.addEdge(edgeAhora);
		sigInstance.refresh();
}

function contains(existingNodes,nodeCont)
{
	var esCierto;
	if(existingNodes.length == 0)
	{
		esCierto = false;
		existingNodes.push(nodeCont.id);
	}
	else
	{
		for(var h=0;h<existingNodes.length;h+=1)
		{
			if(existingNodes[h].id == nodeCont.id)
			{
				esCierto = true;
				break;
			}
			else
				esCierto = false;
		}
		if(esCierto == true)
		{
		}
		else
		{
			esCierto = false;
		}
	}
	return esCierto;
}

function findNodes(nodes,idNodeAct)
{
	var nodeAct;
	for(var j=0; j<nodes.length;j+=1)
	{
		if(idNodeAct == nodes[j].id)
		{
			nodeAct = nodes[j];
		}
	}
	return nodeAct;
}`

**Comment by nelsonmau** [comment]
https://github.com/jacomyal/sigma.js/tree/master/plugins/sigma.parsers.json
